### PR TITLE
Fix SSL check for proxy and redpanda

### PR DESCRIPTION
**Pull request by gbecan**

### CODE REVIEW

**Comment by gbecan** [comment]
Tested on cs-playground (local-with-ldap + local-redpanda-no-security) and several clusters from staging.

**Review by trobert — APPROVED** [review]

File: zio-kafka/src/main/scala/zio/kafka/utils/SslHelper.scala [hunk 1]
```diff
@@ -40,13 +43,10 @@ object SslHelper {
                                     ZIO.attempt(SocketChannel.open(addr))
                                   )(channel => ZIO.attempt(channel.close()).orDie)
                        tls <- ZIO.attempt {
```
Comment (guizmaii) [inline zio-kafka/src/main/scala/zio/kafka/utils/SslHelper.scala:45]: 🌕 I'm wondering if we shouldn't use `attemptBlocking` here. The `channel.read(buf)`call  is most likely blocking while waiting for/receiving the response, no? 🤔

Same for the `SocketChannel.open(addr)` and `channel.close()` call above, these Java functions are most likely blocking IMO. 🤔

**Review by guizmaii — APPROVED** [review]
Really interesting code!

File: zio-kafka/src/main/scala/zio/kafka/utils/SslHelper.scala (hunk 1 above)
Comment (gbecan) [reply zio-kafka/src/main/scala/zio/kafka/utils/SslHelper.scala:45]: The code is already embedded in a `ZIO.blocking` block.

Comment (guizmaii) [reply zio-kafka/src/main/scala/zio/kafka/utils/SslHelper.scala:45]: My bad. Thanks!

**Pull request merged**